src/components/confirm/confirm.pw.tsx

```diff
@@ -0,0 +1,648 @@
+        mount,
+        page,
+      }) => {
+        await mount(
+          <ConfirmComponent
+            cancelButtonIconType="bin"
+            cancelButtonIconPosition={position}
+          />
+        );
+
+        if (position === "before") {
+          const iconPos = page
+            .getByRole("button")
+            .filter({ hasText: "No" })
+            .locator("span")
+            .nth(0);
+          await expect(iconPos).toHaveAttribute("data-element", "bin");
+        } else {
+          const iconPos = page
+            .getByRole("button")
+            .filter({ hasText: "No" })
+            .locator("span:nth-child(2)");
+          await expect(iconPos).toHaveAttribute("data-element", "bin");
+        }
+      });
+    }
+  );
+});
+
+test.describe("should render Confirm component for event tests", () => {
+  test(`should call onCancel callback when a click event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ConfirmComponent
+        onCancel={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const button = page.getByRole("button").filter({ hasText: "No" });
+    await button.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should call onConfirm callback when a click event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ConfirmComponent
+        onConfirm={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    const button = page.getByRole("button").filter({ hasText: "Yes" });
+    await button.click();
+    expect(callbackCount).toBe(1);
+  });
+
+  test(`should check onCancel callback when Esc key event is triggered`, async ({
+    mount,
+    page,
+  }) => {
+    let callbackCount = 0;
+    await mount(
+      <ConfirmComponent
+        onCancel={() => {
+          callbackCount += 1;
+        }}
+      />
+    );
+
+    await page.keyboard.press("Escape");
+    expect(callbackCount).toBe(1);
+  });
+});
+
+test.describe("should check accessibility for Confirm", () => {
+  ([
+    "primary",
+    "secondary",
+    "tertiary",
+    "dashed",
+    "darkBackground",
+  ] as ConfirmProps["confirmButtonType"][]).forEach((type) => {
+    test(`should check accessibility for confirm button of ${type} type`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent confirmButtonType={type} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  ([
+    "primary",
+    "secondary",
+    "tertiary",
+    "dashed",
+    "darkBackground",
+  ] as ConfirmProps["cancelButtonType"][]).forEach((type) => {
+    test(`should check accessibility for cancel button of ${type} type`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<ConfirmComponent cancelButtonType={type} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  ([
```

Comment:
You could use the `disabledStates` array for these tests if you follow the above suggestion